### PR TITLE
Read SBML metabolites and get info from Pubchem

### PR DESCRIPTION
**Pull request by Porthmeus**
I wrote a function which reads an SBML file and retrieves all the information which is available for the metabolites in it and returns a pandas.DataFrame.
Further I created a function which takes a list of common names for metbabolites and retrieves the available information from Pubchem - currently there is 1 second sleep between each request, thus it is not working for large amounts of metabolites.
In the same file you can also find a function, which determines, whether a string is most likely a sum formula of a chemical, which might be helpful in the future.

### CODE REVIEW

**Comment by Porthmeus** [comment]
OK obviously there is no pandas available, however I put it as a requirement in the conda environment. @Unaimend do you know, whether I have to first push the new requirements.yml file to master before I can push more code depending on the new requirements?

**Comment by Unaimend** [comment]
I already looked into it and the environment should contain pandas.
See the logs here.https://github.com/Porthmeus/MeMoMe/actions/runs/4261104460/jobs/7415126504

Tl:dr I don't know why it fails I will look into it today.

**Comment by Unaimend** [comment]
image

File: src/downloadPubchemInfo.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+def downloadElements() -> pd.DataFrame:
+    '''
+    Download the elements table from pubchem and return it as a data frame
+    '''
+
+    # download the element list from pubchem
+    elements = pd.read_csv("https://pubchem.ncbi.nlm.nih.gov/rest/pug/periodictable/CSV?response_type=save&response_basename=PubChemElements_all")
+    return(elements)
+
+def looksLikeSumFormula(string:str, atoms:list = atoms_list) -> bool:
```
Comment (Unaimend) [inline src/downloadPubchemInfo.py:29]: Please include the type of the List i.e. it should be list[str]
See https://docs.python.org/3/library/typing.html for more Informatiom

File: src/downloadPubchemInfo.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    '''
+    Takes a string and tries to determine if it is likely a sum formula
```
Comment (Unaimend) [inline src/downloadPubchemInfo.py:31]: This will fail for stuff like C2H4O2Zn+2 but I assume we only have a subset of possible chemical formulas to check?

**Review by Unaimend — APPROVED** [review]

**Comment by Unaimend** [comment]
@Porthmeus You can check my comments if you want otherwise I think its ready to be merged

File: src/downloadPubchemInfo.py (hunk 2 above)
Comment (Unaimend) [reply src/downloadPubchemInfo.py:31]: Is there some definition of the Pubchem chemical formulas? Maybe a Backus-Naur-Form

Comment (Porthmeus) [reply src/downloadPubchemInfo.py:31]: Yes there is - however not all users comply to it. There is a good chance that the code will fail with common words already, thus I see the need to improve here maybe. However, currently it is used only during a fallback if pubchem can not find an entry to the name given - than I try to remove these formulas, if they are part of the name. I will for now, leave the code as it is and put it as an improvement to the issues.

Comment (Porthmeus) [reply src/downloadPubchemInfo.py:31]: Backus-Naur would look something like [<Atom>|<NumberOfTime>]n

**Pull request merged**